Cpp-fy libsweep/protocol

MikeGitb opened the pull request:
A couple of small refactorings that make protocol.h and protocol.cc a bit more "cpp-ish"

MikeGitb commented:
Case of bad timing, as it probably conflicts with #80.
If merging turns out to be non-trivial, I'd be happy to rebase onto the result of #80

daniel-j-h commented on libsweep/include/protocol.h:
```diff
@@ -6,7 +6,7 @@
  */
 
-#include <stdint.h>
+#include <cstdint>
```
I never fully understood the difference here - I thought if we want to use `int32_t` (unprefixed) we have to include `stdint.h`, and if we include `cstdint` we _have to_ use `std::int32_t` to be on the safe side?

MikeGitb replied:
@daniel-j-h: I think you are right <cstdint> only guarantees the availability of `std::int32_t` not the un-prefixed version.

daniel-j-h commented on libsweep/include/protocol.h:
```diff
@@ -206,8 +206,7 @@ inline void integral_to_ascii_bytes(const int32_t integral, uint8_t bytes[2]) {
 
-  // Numbers begin at ASCII code point 48
-  const uint8_t ASCIINumberBlockOffset = 48;
+  const uint8_t ASCIINumberBlockOffset = '0';
```
Line 197 does the same and should probably adapted, too.

MikeGitb replied:
Right, overlooked that.

MikeGitb commented:
Ok, I think I fixed your comments: I left the stdint.h header in place and removed  the two other ones that appered to be no longer in use anyway.
 
Regarding the merge order, do whatever you think is easiest. when in doubt I'd give priority to #80 as this one here makes mostly cosmetic changes. It was actually meant to be the first step of multiple PRs (ironically also introducing exceptions).
